source/includes/rest/_batch_api.md

```diff
@@ -326,3 +326,189 @@ request and the time your data appears in user interface.
     </tr>
   </tbody>
 </table>
+
+## Create or update a batch of orders
+
+> To create or update a batch of orders:
+
+```shell
+curl -X POST "https://api.getdrip.com/v2/YOUR_ACCOUNT_ID/subscribers/batches" \
+  -H 'User-Agent: Your App Name (www.yourapp.com)' \
+  -u YOUR_API_KEY: \
+  -d @- << EOF
+  {
+    "batches": [{
+      "orders": [
+        {
```

Comment:
What's the logic behind having an array of batches? I see that that's how all our Batch APIs are implemented, but I couldn't find anywhere in the docs that we explain it.

Reply:
Ok, I now see the params limit to 1k, so are we saying that a user could send as much data as they want in a single request as long as the user splits it into batches of 1000? I'd say we should just drop that array (at least for v3) and limit the user to 1k per API call, not 1k per batch and (unlimited?) batches per API call. For now, this is probably OK.

This also raises another interesting question—what do we do with our API Docs as we switch over to v3?

Reply:
Ok, and upon investigating the controller itself, I'm now even more convinced we have an issue here. We actually DO only accept a single batch, but there's nothing in the documentation that states that, and no error raised if you send multiple batches. Instead, it looks like we simply ignore all batches after the first?

Reply:
We could either correct the docs, or raise an error, or even better, both, in that situation. Downside to raising an error would be we'd also drop the otherwise accepted part of the payload. 

A third option would be to update our API to accept multiple batches in a single request, but that leaves this endpoint open to people dumping TBs of data in a single request.

Reply:
The pattern for we use batches is the same for the other parts of the REST API. We always expect a single resource wrapped in an array. e.g. to update a subscriber, you send a `subscribers` array with one object containing subscriber data:

```
curl -X POST "https://api.getdrip.com/v2/YOUR_ACCOUNT_ID/subscribers" \
  -H 'User-Agent: Your App Name (www.yourapp.com)' \
  -u YOUR_API_KEY: \
  -d @- << EOF
  {
    "subscribers": [{
      "email": "john@acme.com",
      "time_zone": "America/Los_Angeles",
      "custom_fields": {
        "name": "John Doe"
      }
    }]
  }
  EOF
```

That endpoint never accepts more than one member of the `subscriber` array. That's the pattern we are using for `batches`. We never accept more than one batch. It's based on an old version of the JSON API spec: http://jsonapi.org/.

We could make that clearer in the docs.

Reply:
Maybe the docs isn't the right place, maybe we should update the API to raise an error if someone tries to send more data than we allow. For instance, if someone sends 2 subscribers at once, today we'll drop the second subscriber and return a `:created` status after only creating the first subscriber. I'd argue that we would be better off raising an error so that the user knows their data could not be accepted as provided, and require them to resubmit exactly 1.

Reply:
That sounds reasonable to me. Should we create a separate issue for this? We'd need to cover all the places where we use this pattern for creating resources: subscribers, events, orders, campaign subscriptions, workflow subscriptions, workflow triggers, + anything I'm forgetting.

Reply:
I created that separate issue. #8203